src/scenic/syntax/compiler.py

```diff
@@ -476,6 +479,63 @@ def visit_MonitorDef(self, node: s.MonitorDef):
             body=node.body,
         )
 
+    def visit_ScenarioDef(self, node: s.ScenarioDef):
+        # TODO(shun): assert not in behavior or compose
+
+        # Set up arguments for setup and compose blocks
+        args: ast.arguments = self.visit(node.args)
+        args.posonlyargs = initialBehaviorArgs + args.posonlyargs
+
+        # Get preconditions and invariants
+        preconditions, invariants = self.separatePreconditionsAndInvariants(node.header)
+
+        setup = node.setup
+        compose = node.compose
+
+        # Find all locals of the scenario, which will be shared amongst the various blocks
+        allLocals = set()
+        if node.compose:
+            allLocals.update(LocalFinder.findIn(node.compose))
+        if node.setup:
+            allLocals.update(LocalFinder.findIn(node.setup))
+        oldBL = self.behaviorLocals
+        self.behaviorLocals = allLocals
+
+        guardCheckers = self.makeGuardCheckers(args, preconditions, invariants)
```

Comment:
Looks like this line got duplicated; only the copy on line 508 should exist.

Reply:
Thanks for catching this. Fixed in 6f97e73e275f706e0910413aee4430e9a1c20748